src/core/Akka.Streams.Tests/Dsl/RepeatPreviousSpec.cs

```diff
@@ -0,0 +1,84 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="RepeatPreviousSpec.cs" company="Akka.NET Project">
+//     Copyright (C) 2013-2022 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Akka.Configuration;
+using Akka.Streams.Dsl;
+using Akka.TestKit;
+using FluentAssertions;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Akka.Streams.Tests.Dsl
+{
+    public class RepeatPreviousSpec : AkkaSpec
+    {
+        private ActorMaterializer Materializer { get; }
+
+        public RepeatPreviousSpec(ITestOutputHelper testOutputHelper) : base(Config.Empty, output: testOutputHelper)
+        {
+            var settings = ActorMaterializerSettings.Create(Sys);
+            Materializer = ActorMaterializer.Create(Sys, settings);
+        }
+        
+        [Fact]
+        public async Task RepeatPrevious_should_immediately_terminate_with_Empty_source()
+        {
+            var source = Source.Empty<int>();
+            var result = await source.RepeatPrevious().RunWith(Sink.Seq<int>(), Materializer);
+            result.Should().BeEmpty();
+        }
+        
+        [Fact]
+        public async Task RepeatPrevious_should_complete_when_upstream_completes()
+        {
+            var source = Source.Single(1).RepeatPrevious();
+            var result = await source.RunWith(Sink.Seq<int>(), Materializer);
+
+            // as a side-effect of RepeatPrevious' buffering process, there's going to be an extra element in the result
+            result.Should().BeEquivalentTo(1, 1);
```

Comment:
Unfortunately, and I spent a couple of hours trying to fix this - we're going to get an extra event due to the work involved in detecting that the upstream has completed. Comes with the territory if you're separating upstream from downstream in the manner that this stage does.

Reply:
This comment may help: https://github.com/akkadotnet/akka.net/pull/6262#discussion_r1031828989